modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/tasks/WatchTask.java

```diff
@@ -355,6 +364,33 @@ private static String _sendGogoShellCommand(
 		return response;
 	}
 
+	private String _getFragmentHostName() throws IOException {
+		String retVal = null;
+
+		FileCollection fileCollection = getProject().files("bnd.bnd");
+
+		if (fileCollection != null) {
+			Set<File> files = fileCollection.getFiles();
+
+			for (File file : files) {
```

Comment:
No need to iterate over the files.  Just do this `fileCollection.getSingleFile()`